Update edit link page

brenetic opened the pull request:
- The edit link page now has both the service title and the interaction
title
- 'LGSL code' is now 'Service code'
- Removed LGIL
- 'URL' is now 'Page URL'
- Removed the ability to edit the homepage url
- 'Save' is now 'Update'
- Confirmation before deletion message
- Return to correct page upon update, cancel and delete of a link

# Before

<img width="1172" alt="before" src="https://cloud.githubusercontent.com/assets/3466862/20602414/f584cfee-b254-11e6-8e5d-1720cae02bc3.png">

# After

<img width="1183" alt="after" src="https://cloud.githubusercontent.com/assets/3466862/20602823/d477defc-b256-11e6-8e16-237b90d8361b.png">

<img width="1170" alt="screen shot 2016-11-25 at 12 34 47" src="https://cloud.githubusercontent.com/assets/3466862/20625494/a0b30244-b30b-11e6-97bc-c44010b51cce.png">


[Trello card](https://trello.com/c/gh6xcdeY/564-update-edit-links-and-add-link-pages-2)

brenetic committed:
Update edit link page

- The edit link page now has both the service title and the interaction
title
- 'LGSL code' is now 'Service code'
- Removed LGIL
- 'URL' is now 'Page URL'
- Removed the ability to edit the homepage url
- 'Save' is now 'Update'
- 'eg' is now 'for example'

[Trello card](https://trello.com/c/gh6xcdeY/564-update-edit-links-and-add-link-pages-2)

issyl0 commented on app/views/links/edit.html.erb:
```diff
@@ -1,30 +1,32 @@
+      <%= f.label :url, 'Page URL' %>
 
-      <p class="text-muted">eg http://www.local-authority.gov.uk</p>
+      <p class="text-muted">example http://www.local-authority.gov.uk</p>
```
ISTR this should be "for example", otherwise LGTM pending the PR it relies upon being merged.

brenetic committed:
Add redirect to correct previous page on save, cancel and delete

brenetic committed:
Add warning pop up when deleting a link

brenetic committed:
Change render edit to redirect

This will allow users to use their browser's back button and not get a
"Confirm form resubmission" message

whoojemaflip commented on app/controllers/links_controller.rb:
```diff
@@ -46,9 +51,20 @@ def load_dependencies
+    redirect_to edit_link_path(@local_authority, @service, @interaction)
+  end
+
   def redirect(action = 'saved')
```
Could this be called `redirect_back` to give a clearer idea of what it's doing?

whoojemaflip commented on app/views/interactions/index.html.erb:
```diff
@@ -25,7 +25,12 @@
         <% next if interaction.link_url.nil? %>
-        <tr id="<%= interaction.lgil_code %>" <% if flash[:lgil] == interaction.lgil_code %>class="success"<% end %>>
+        <tr id="<%= interaction.lgil_code %>"
+        <% if flash[:updated].present? &&
```
if only there were some kind of Presenter to delegate this logic to...

whoojemaflip commented on app/views/links/edit.html.erb:
```diff
@@ -25,7 +25,7 @@
 
     <div class='actions add-vertical-margins col-xs-6 no-gutter'>
-      <%= link_to 'Cancel', local_authority_with_service_path(local_authority_slug: @local_authority.slug, service_slug: @service.slug), class: 'btn btn-default add-right-margin' %>
+      <%= link_to 'Cancel', flash[:back], class: 'btn btn-default add-right-margin' %>
```
instead of using `flash[:back]` here, would it be possible to use a helper method to wrap this functionality, and provide a fallback?

whoojemaflip commented on app/controllers/links_controller.rb:
```diff
@@ -46,6 +54,11 @@ def load_dependencies
   end
 
+  def redirect_on_failed_validation
+    flash[:link_url] = params[:link][:url].strip
```
There's some duplication of `params[:link][:url].strip` - could this be extracted into a method?

whoojemaflip commented on app/presenters/url_status_presentation.rb:
```diff
@@ -23,4 +23,10 @@ def last_checked
     end
   end
+
+  def updated_url?
```
It might be more idiomatic to call this `updated?`

whoojemaflip commented on spec/features/local_authorities/local_authority_show_spec.rb:
```diff
@@ -97,6 +97,24 @@
+        click_on('Update')
+
+        expect(page).to have_text("Services and links")
+        expect(page).to have_link('http://angus.example.com/link-to-change')
```
Would this test pass if the page just reloaded?  What about an assertion on the URL?

whoojemaflip commented on app/controllers/links_controller.rb:
```diff
@@ -1,20 +1,15 @@
 
 class LinksController < ApplicationController
-  before_action :load_dependencies
+  before_action :load_dependencies, :set_back_url_before_post_request
```
is this relevant to all controller methods, or could it be restricted to just one or two?

brenetic commented:
@whoojemaflip  Addressed all the comments and did most of the refactor for the `Links#index`. Unfortunately one set of failing tests remain . . . 💥 🔥 👎

issyl0 commented:
Are the tests failing because there's a trailing `:` in `(def initialize(link, view_context:)` that it then can't find `view_context` without the `:` in the method itself? *shrug*

EDIT: Nope, that didn't help at all in dev. Confused (and I'd never seen that pattern before).

davidbasalla commented:
Could we split that out that last commit into a different PR? It seems like a substantial change which is not directly related to the 'edit link' page

brenetic commented:
@issyl0 the `view_context` is a named parameter

brenetic commented:
@davidbasalla  Removed the last commit. Will raise another PR

whoojemaflip commented on app/controllers/links_controller.rb:
```diff
@@ -2,41 +2,48 @@
 
 class LinksController < ApplicationController
   before_action :load_dependencies
+  before_action :set_back_url_before_post_request, only: [:edit, :update, :destroy]
```
this might be more succinctly expressed as
`before_action :set_back_url, if: -> { request.post? }`

NBD though, only change it if you want to.

whoojemaflip reviewed (approved):
Cool cool cool
